pkg/security/probe/probe.go

```diff
@@ -1043,6 +1045,13 @@ func NewProbe(config *config.Config, client *statsd.Client) (*Probe, error) {
 	return p, nil
 }
 
+func (p *Probe) sanityChecks() {
+	// enable runtime compiled constants on COS by default
+	if !p.config.RuntimeCompiledConstantsIsSet && p.kernelVersion.IsCOSKernel() {
```

Comment:
Confidence check: are you forced to test for `!p.config.RuntimeCompiledConstantsIsSet`?

Reply:
The idea is: we want the default value to be true only on Container Optimized OS. The problem is we don't know if we are on COS when defining the default values in `config.go`.

Reply:
Yes exactly